### PR TITLE
WIP: ENH: basic commands

### PR DESCRIPTION
**Pull request by bpoldrack**
Implementation of the first set of basic datalad commands.

Attention: Depends on #157 ! 
No merging yet!

### CODE REVIEW

**Comment by hanke** [comment]
I get this

% PYTHONPATH=~/hacking/datalad/git ~/hacking/datalad/git/bin/datalad add-handle ~/forrest/anondata .
2015-08-20 13:44:56,155 [ERROR  ] 'Namespace' object has no attribute 'h_path' (AttributeError) (main.py:186)

Moreover, I think that the collection path could be optional for add-handle, if the curdir is already a collection repo. If not, and it is not given: either fail or add to the local collection (not sure).

**Comment by hanke** [comment]
Another thought: we need to (somehow) group command. Even with the few we have it is already getting incomprehensive which are collection/handle/other-related.

**Comment by hanke** [comment]
@bpoldrack will draft something for command grouping

File: datalad/interface/create_collection.py [hunk 1]
```diff
@@ -12,17 +12,44 @@
...
+from os import curdir
+from os.path import join as opj, abspath, expandvars, expanduser
 from .base import Interface
 from datalad.support.param import Parameter
-from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureStr, EnsureNone
+from datalad.support.collectionrepo import CollectionRepo
+from appdirs import AppDirs
+
+dirs = AppDirs("datalad", "datalad.org")
```
Comment (yarikoptic) [inline datalad/interface/create_collection.py]: I see this pattern over and over again -- I guess we need some kind of 'datalad.dirs' with some helpers like `get_user_data_path(filename)` which would do the magical import, AppDirs, opj?

Comment (bpoldrack) [inline datalad/interface/create_collection.py]: Right. And there are/will be more things like that. Just not sure where to
put it best yet.

2015-09-04 19:02 GMT+02:00 Yaroslav Halchenko <notifications@github.com>:

**Pull request merged**